File: backend/app/schemas/shipping.py
from pydantic import BaseModel
from typing import List, Optional
from app.models.shipping_record import ShippingStatus


class ShippingRecordOut(BaseModel):
    id: int
    issue_id: int
    recipient_id: int
    recipient_name: str
    recipient_address: Optional[str]
    recipient_phone: Optional[str]
    recipient_type: str
    quantity: int
    status: ShippingStatus

    model_config = {"from_attributes": True}


class ShippingRecordUpdate(BaseModel):
    recipient_id: int
    quantity: int


class ShippingDataUpdate(BaseModel):
    records: List[ShippingRecordUpdate]

File: frontend/src/api/shipping.ts
import api from './client';

export interface ShippingRecord {
  id: number;
  issue_id: number;
  recipient_id: number;
  recipient_name: string;
  recipient_address: string | null;
  recipient_phone: string | null;
  recipient_type: string;
  quantity: number;
  status: string;
}

export const getShipping = (issueId: number) =>
  api.get<ShippingRecord[]>(`/issues/${issueId}/shipping`);

export const regenerateShipping = (issueId: number) =>
  api.post<ShippingRecord[]>(`/issues/${issueId}/shipping/regenerate`);
